src/Lib/Managers/ExportAreaManager.vala

```diff
@@ -0,0 +1,287 @@
+
+    public ExportAreaManager (Akira.Lib.Canvas canvas) {
+        Object (
+            canvas: canvas
+        );
+    }
+
+    public Goo.CanvasRect create_area (Gdk.EventButton event) {
+        var dash = new Goo.CanvasLineDash (2, 5.0, 5.0);
+        var rgba_fill = Gdk.RGBA ();
+        rgba_fill.parse (COLOR);
+        rgba_fill.alpha = 0.1;
+        uint fill_color_rgba = Utils.Color.rgba_to_uint (rgba_fill);
+
+        area = new Goo.CanvasRect (
+            null,
+            Utils.AffineTransform.fix_size (event.x),
+            Utils.AffineTransform.fix_size (event.y),
+            1.0, 1.0,
+            "line-width", LINE_WIDTH,
+            "stroke-color", COLOR,
+            "line-dash", dash,
+            "fill-color-rgba", fill_color_rgba,
+            null
+        );
+
+        initial_x = event.x;
+        initial_y = event.y;
+        initial_width = 1.0;
+        initial_height = 1.0;
+
+        area.set ("parent", canvas.get_root_item ());
+        area.can_focus = false;
+
+        return area;
+    }
+
+    public void resize_area (double x, double y) {
+        canvas.convert_to_item_space (area, ref x, ref y);
+
+        double delta_x = x - initial_x;
+        double delta_y = y - initial_y;
+
+        double item_width = area.width;
+        double item_height = area.height;
+
+        double new_width = item_width;
+        double new_height = item_height;
+
+        double origin_move_delta_x = 0.0;
+        double origin_move_delta_y = 0.0;
+
+        new_width = initial_width + delta_x;
+        new_height = initial_height + delta_y;
+        if (canvas.ctrl_is_pressed && new_height > MIN_SIZE) {
+            new_height = new_width;
+        }
+
+        if (new_width < initial_width) {
+            new_width = initial_width - delta_x;
+            origin_move_delta_x = item_width - new_width;
+        }
+
+        if (new_height < MIN_SIZE) {
+            new_height = initial_height - delta_y;
+            origin_move_delta_y = item_height - new_height;
+        }
+
+        new_width = Utils.AffineTransform.fix_size (new_width);
+        new_height = Utils.AffineTransform.fix_size (new_height);
+        origin_move_delta_x = Utils.AffineTransform.fix_size (origin_move_delta_x);
+        origin_move_delta_y = Utils.AffineTransform.fix_size (origin_move_delta_y);
+
+        canvas.convert_from_item_space (area, ref initial_x, ref initial_y);
+        area.translate (origin_move_delta_x, origin_move_delta_y);
+        canvas.convert_to_item_space (area, ref initial_x, ref initial_y);
+
+        Utils.AffineTransform.set_size (new_width, new_height, area);
+    }
+
+    public void clear () {
+        if (area == null) {
+            return;
+        }
+
+        area.remove ();
+    }
+
+    public void create_export_snapshot () {
+        // Hide the area before rendering.
+        area.visibility = Goo.CanvasItemVisibility.INVISIBLE;
+        // Generate the image to export.
+        generate_pixbuf ();
```

Comment:
Maybe this could be an improvement for next versions, not a bug. Since the `generate_pixbuf' could take a while to finish and in the meantime the canvas will be "freezed", I suggest to create and show the dialog beforehand and then to show a progress indicator (or simply a loader) so inform the user some action is taking place. 

Once the pixbuf generation has been done, the user is presented with the image and then he/she can perform other adjustments and could see the results in the window. 

The same can be said for when the user click on the Export button, right now there's no confirmation that the file has been correctly saved, so we could/should provide a screen to inform the user the action has been completed and that the newly created image is ready. 

Again, all those things can be accounted for in a future PR, but it seemed to me that this could be a good place to start this conversation. 

What do you think?

Reply:
Yes, these are very good points.
I think I will update all these method to be `async`, and add some multi threading when necessary, but in a follow up bug.

This is mostly the first barebone implementation of a single export, but once we start dealing with big artboards or multiple exports at once, progress bars and confirmation dialogs are definitely necessary.

Reply:
Nevermind, I implemented async methods and multithreading to deal with it right away :smile: